From 4d90cb24a53cd7d5776d9b902e04cfee37305c22 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Wed, 31 Jul 2024 15:47:45 +0200
Subject: [PATCH 1070/1695] dts: stm32: Fix ranges and dma-range property of
 mlahb node on stm32mp23

The parent node of mlahb is the root node which fixes size-cells to 2.
this implies that the bus address must de declared in 64-bit format.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c4216dc7fdd32ddc3a9c5c7f0e6ec896dc2c62df dts: stm32: Fix ranges and dma-range property of mlahb node on stm32mp23]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I05e05b7bb3ec44166fefe7f223d9b1ab4cc1d711
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/396246
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 core/arch/arm/dts/stm32mp231.dtsi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index dd1f0cfad..a4ee0bae7 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -121,8 +121,8 @@
 		compatible = "st,mlahb", "simple-bus";
 		#address-cells = <1>;
 		#size-cells = <1>;
-		ranges = <0x0 0x0  0xFFFFFFFC>;
-		dma-ranges = <0x0 0x0 0xFFFFFFFC>;
+		ranges = <0x0 0x0 0x0 0xFFFFFFFC>;
+		dma-ranges = <0x0 0x0 0x0 0xFFFFFFFC>;
 
 		m33_rproc: m33@0 {
 			compatible = "st,stm32mp2-m33";
-- 
2.47.2

